devenv/src/cli.rs

```diff
@@ -63,15 +63,19 @@ pub struct GlobalOptions {
     pub log_format: LogFormat,
 
     #[arg(short = 'j', long,
-        global = true, help = "Maximum number of Nix builds at any time.",
-        default_value_t = max_jobs())]
+        global = true,
+        env = "DEVENV_MAX_JOBS",
```

Comment:
I'm not sure we should allow env variables here.

We should think about a global user config, because this is about the machine we're running on.